### PR TITLE
feat(rule): Adds rule to disallow React Hooks

### PR DESCRIPTION
**Pull request by matejminar**
Rule to disallow React Hooks as we discussed on our TSC.

### CODE REVIEW

**Comment by markstory** [comment]
Nice work @matejminar :clap:

File: packages/eslint-config-sentry-app/strict.js [hunk 1]
```diff
@@ -40,6 +40,8 @@ module.exports = {
 
     'sentry/no-styled-shortcut': ['error'],
 
+    'sentry/no-react-hooks': ['error'],
```
Comment (billyvg) [inline packages/eslint-config-sentry-app/strict.js]: Let's move this to the default ruleset since we don't need to change any code for it.

**Comment by billyvg** [comment]
Looks good! Thanks for doing this!

File: packages/eslint-config-sentry-app/strict.js (hunk 1 above)
Comment (matejminar) [reply packages/eslint-config-sentry-app/strict.js]: 👍

**Comment by matejminar** [comment]
Thanks, it was my pleasure. I've moved the rule from strict to default, so I think we are ready to merge&publish.

**Pull request merged**